test/performance/benchmarks/dataplane-probe/main.go

```diff
@@ -0,0 +1,188 @@
+	ctx, startInformers := injection.EnableInjectionOrDie(ctx, cfg)
+	startInformers()
+
+	if *target == "" {
+		log.Fatalf("-target is a required flag.")
+	}
+
+	log.Println("Starting dataplane probe for target:", *target)
+
+	ctx, cancel := context.WithTimeout(ctx, *duration+time.Minute)
+	defer cancel()
+
+	// Based on the "target" flag, load up our target benchmark.
+	// We only run one variation per run to avoid the runs being noisy neighbors,
+	// which in early iterations of the benchmark resulted in latency bleeding
+	// across the different workload types.
+	t, ok := targets[*target]
+	if !ok {
+		log.Fatalf("Unrecognized target: %s", *target)
+	}
+
+	// Make sure the target is ready before sending the large amount of requests.
+	if err := performance.ProbeTargetTillReady(t.target.URL, *duration); err != nil {
+		log.Fatalf("Failed to get target ready for attacking: %v", err)
+	}
+
+	// Send 1000 QPS (1 per ms) for the given duration with a 30s request timeout.
+	rate := vegeta.Rate{Freq: 1, Per: time.Millisecond}
+	targeter := vegeta.NewStaticTargeter(t.target)
+	attacker := vegeta.NewAttacker(vegeta.Timeout(30 * time.Second))
+
+	influxReporter, err := performance.NewInfluxReporter(map[string]string{"target": *target})
+	if err != nil {
+		log.Fatalf("failed to create influx reporter: %v", err.Error())
+	}
+	defer influxReporter.FlushAndShutdown()
+
+	// Start the attack!
+	results := attacker.Attack(targeter, rate, *duration, "load-test")
+	deploymentStatus := performance.FetchDeploymentStatus(ctx, system.Namespace(), "activator", time.Second)
+
+	var metricResults vegeta.Metrics
+
+LOOP:
+	for {
+		select {
+		case <-ctx.Done():
+			// If we time out or the pod gets shutdown via SIGTERM then start to
+			// clean thing up.
+			break LOOP
+
+		case ds := <-deploymentStatus:
+			// Report number of ready activators.
+			influxReporter.AddDataPoint(benchmarkName, map[string]interface{}{"activator-pod-count": float64(ds.ReadyReplicas)})
+
+		case res, ok := <-results:
+			if ok {
+				metricResults.Add(res)
+			} else {
+				// If there are no more results, then we're done!
+				break LOOP
+			}
+		}
+	}
+
+	// Compute latency percentiles
+	metricResults.Close()
+
+	// Report results to influx
+	influxReporter.AddDataPoint(benchmarkName, map[string]interface{}{"requests": float64(metricResults.Requests)})
+	influxReporter.AddDataPoint(benchmarkName, map[string]interface{}{"latency-mean": float64(metricResults.Latencies.Mean)})
+	influxReporter.AddDataPoint(benchmarkName, map[string]interface{}{"latency-min": float64(metricResults.Latencies.Min)})
+	influxReporter.AddDataPoint(benchmarkName, map[string]interface{}{"latency-max": float64(metricResults.Latencies.Max)})
+	influxReporter.AddDataPoint(benchmarkName, map[string]interface{}{"latency-p95": float64(metricResults.Latencies.P95)})
+	influxReporter.AddDataPoint(benchmarkName, map[string]interface{}{"errors": float64(len(metricResults.Errors))})
```

Comment:
How about re-writing to: 

```
	metrics := []map[string]interface{}{
		{
			"requests": float64(metricResults.Requests),
		}, 
....
	}
	
	for _, m := range metrics {
		influxReporter.AddDataPoint(benchmarkName, m)
	}

```

Also we could add this to a func and call it add different places as it seems multiple main funcs report the same metrics.